Command Line Args

Hack5190 opened the pull request:
added code to parse command line options "-t token-filename", "-i IP-address", "-p port" and "-u no_userlist"

added code to support the use of "-i" and "-p"

ltworf commented on irc.py:
```diff
@@ -257,13 +260,52 @@ def command(self, cmd: bytes) -> None:
+    uuser=''
+ 
+    # Read command line args
+    myopts, args = getopt.getopt(sys.argv[1:],"t:i:p:u:")
```
Can you have a look at the `argparse` module for reading command line? 

It's more high level. getopt is more or less like in C. That one can have help strings in the parameters already, enforce types, set default values.

ltworf commented on irc.py:
```diff
@@ -257,13 +260,52 @@ def command(self, cmd: bytes) -> None:
+            print("Usage: %s -t token-filename -i IP-address -p port-number -u no-userlist" % sys.argv[0])
+ 
+    nuip=ip2long(uip)
+    uip=socket.inet_ntoa(struct.pack('!L', nuip))
```
no need in python, you can just pass the string `127.0.0.1` as I do and it works.

Hack5190 commented:
ltworf,

Thanks for the tip on the IP address, I removed the unneeded code from my
local version.  Because I don't know Python each change requires a fair
amount of time to research and test.  Right now I just don't have the spare
time to replace code that works.  If you want to replace getopt.getopt with
argparse that would be great.  Today I got the -u option working and plan
to continue with the changes I proposed in the hope I'll be able to help
with the project.

Hack5190

On Mon, Apr 30, 2018 at 5:58 AM ltworf <notifications@github.com> wrote:

> *@ltworf* commented on this pull request.
> ------------------------------
>
> In irc.py
> <https://github.com/ltworf/localslackirc/pull/21#discussion_r184948606>:
>
> >
>  def main():
> +    ###############################
> +    # t == token-filename
> +    # i == IP-Address
> +    # p == port-number
> +    # u == no-userlist
> +    ###############################
> +    utoken=''
> +    uip='127.0.0.1'
> +    uport='9007'
> +    uuser=''
> +
> +    # Read command line args
> +    myopts, args = getopt.getopt(sys.argv[1:],"t:i:p:u:")
>
> Can you have a look at the argparse module for reading command line?
>
> It's more high level. getopt is more or less like in C. That one can have
> help strings in the parameters already, enforce types, set default values.
> ------------------------------
>
> In irc.py
> <https://github.com/ltworf/localslackirc/pull/21#discussion_r184948699>:
>
> > +    # a == argument passed to the o
> +    ###############################
> +    for o, a in myopts:
> +        if o == '-t':
> +            utoken=a
> +        elif o == '-i':
> +            uip=a
> +        elif o == '-p':
> +            uport=a
> +        elif o == '-u':
> +            uuser=a
> +        else:
> +            print("Usage: %s -t token-filename -i IP-address -p port-number -u no-userlist" % sys.argv[0])
> +
> +    nuip=ip2long(uip)
> +    uip=socket.inet_ntoa(struct.pack('!L', nuip))
>
> no need in python, you can just pass the string 127.0.0.1 as I do and it
> works.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ltworf/localslackirc/pull/21#pullrequestreview-116240983>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP86FYAlP338vUkBqfq3WMKvDy63eBiGks5ttuA0gaJpZM4Tr-8e>
> .
>
-- 
Hack to Live
Live to Hack

My PGP Key is here <https://pastebin.com/Lvd06Sqm> and here
<https://pgp.mit.edu/pks/lookup?op=get&search=0x653466A54731A351>.

Fingerprint=BE52 B2EE 2D39 1240 6B95  77FB 6534 66A5 4731 A351

Hack5190 commented:
On Tue, May 1, 2018 at 03:22 ltworf <notifications@github.com> wrote:

> If you made changes, you can do git push on the same branch, and they end
> up in this pull request.
>
> I won't merge it unless it has argparse tough.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ltworf/localslackirc/pull/21#issuecomment-385615259>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP86FeZDpDbOM6Ct6WS9GFd1WGD6Xotxks5tuA1GgaJpZM4Tr-8e>
> .
>
It’s your project, I fully understand and respect your position.
Unfortunately learning / changing that code just doesn’t work for me due to
other commitments right now.  Hopefully I will still be able help in some
way.

Thanks
-- 
Hack to Live
Live to Hack

ltworf commented on irc.py:
```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python3
+#!/usr/bin/env python3
```
?

ltworf commented on irc.py:
```diff
@@ -41,6 +42,9 @@ def __init__(self, s, sl_client):
         self.sl_client = sl_client
 
+     def _ignorehandler(self, cmd: bytes) -> None:
+         dummy=1
```
the python thing is `pass`

ltworf commented on irc.py:
```diff
@@ -262,6 +266,7 @@ def command(self, cmd: bytes) -> None:
             b'LIST': self._listhandler,
             b'WHO': self._whohandler,
             b'MODE': self._modehandler,
+            b'ISON': self._ignorehandler,
```
What is the point of handling this if the handler does nothing?

Hack5190 commented:
Removed duplicate **python3** line at start of file
Replaced **dummy=1** with **pass**
The purpose of  **b'ISON': self._ignorehandler** is to stop **Message(421): ISON Unknown command** from being repeatedly displayed in the server / network output window

ltworf commented:
Anyway a PR should contain only one set of specific changes, other changes belong in other pull requests, not all in one.

Also this one has conflicts and can't be merged.

For those reasons i closed it.

It is better that you upgrade your fork to the latest master, and make separate pull requests.

Also, daniel is working on the command line thing, so maybe you want to talk on IRC and avoid duplicating work.

Hack5190 commented:
All on the drawing board for some point in the future ….

On Wed, May 2, 2018 at 2:44 PM ltworf <notifications@github.com> wrote:

> Anyway a PR should contain only one set of specific changes, other changes
> belong in other pull requests, not all in one.
>
> Also this one has conflicts and can't be merged.
>
> For those reasons i closed it.
>
> It is better that you upgrade your fork to the latest master, and make
> separate pull requests.
>
> Also, daniel is working on the command line thing, so maybe you want to
> talk on IRC and avoid duplicating work.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ltworf/localslackirc/pull/21#issuecomment-386080136>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP86FdyR2IHAPHDErb5vegL6AvK4KfS0ks5tuf53gaJpZM4Tr-8e>
> .
>


-- 
Hack to Live
Live to Hack

My PGP Key is here <https://pastebin.com/Lvd06Sqm> and here
<https://pgp.mit.edu/pks/lookup?op=get&search=0x653466A54731A351>.

Fingerprint=BE52 B2EE 2D39 1240 6B95  77FB 6534 66A5 4731 A351
